services/wallet/chain/client.go

```diff
@@ -13,22 +13,31 @@ import (
 	"github.com/status-im/status-go/services/rpcstats"
 )
 
+var ChainClientInstances = make(map[uint64]*Client)
+
 type Client struct {
-	eth       *ethclient.Client
-	ChainID   uint64
-	rpcClient *rpc.Client
+	eth         *ethclient.Client
+	ChainID     uint64
+	rpcClient   *rpc.Client
+	IsConnected bool
```

Comment:
Is it possible for a client to be called by goroutines? if so, it might be a good idea to convert the client into a mutex and do  `cc.Lock()` / `cc.RLock()` wherever you set or access the `IsConnected` attribute

```suggestion
        sync.RWMutex

	eth         *ethclient.Client
	ChainID     uint64
	rpcClient   *rpc.Client
	IsConnected bool
```